packages/js/product-editor/src/components/variations-table/variations-table.tsx

```diff
@@ -60,41 +60,56 @@ export function VariationsTable() {
 	const productId = useEntityId( 'postType', 'product' );
 	const context = useContext( CurrencyContext );
 	const { formatAmount } = context;
-	const { isLoading, variations, totalCount } = useSelect(
-		( select ) => {
-			const {
-				getProductVariations,
-				hasFinishedResolution,
-				getProductVariationsTotalCount,
-			} = select( EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME );
-			const requestParams = {
-				product_id: productId,
-				page: currentPage,
-				per_page: perPage,
-				order: 'asc',
-				orderby: 'menu_order',
-			};
-			return {
-				isLoading: ! hasFinishedResolution( 'getProductVariations', [
-					requestParams,
-				] ),
-				variations:
-					getProductVariations< ProductVariation[] >( requestParams ),
-				totalCount:
-					getProductVariationsTotalCount< number >( requestParams ),
-			};
-		},
-		[ currentPage, perPage, productId ]
-	);
+	const { isLoading, variations, totalCount, isGeneratingVariations } =
+		useSelect(
+			( select ) => {
+				const {
+					getProductVariations,
+					hasFinishedResolution,
+					getProductVariationsTotalCount,
+					isGeneratingVariations: getIsGeneratingVariations,
+				} = select( EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME );
+				const requestParams = {
+					product_id: productId,
+					page: currentPage,
+					per_page: perPage,
+					order: 'asc',
+					orderby: 'menu_order',
+				};
+				return {
+					isLoading: ! hasFinishedResolution(
+						'getProductVariations',
+						[ requestParams ]
+					),
+					isGeneratingVariations: getIsGeneratingVariations( {
+						product_id: productId,
+					} ),
+					variations:
+						getProductVariations< ProductVariation[] >(
+							requestParams
+						),
+					totalCount:
+						getProductVariationsTotalCount< number >(
+							requestParams
+						),
+				};
+			},
+			[ currentPage, perPage, productId ]
+		);
 
 	const { updateProductVariation } = useDispatch(
 		EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
 	);
 
-	if ( ! variations || isLoading ) {
+	if ( ! variations && isLoading ) {
 		return (
-			<div className="woocommerce-product-variations is-loading">
+			<div className="woocommerce-product-variations__loading">
 				<Spinner />
+				{ isGeneratingVariations && (
+					<span>
+						{ __( 'Generating variations...', 'woocommerce' ) }
+					</span>
```

Comment:
We are actually able to accurately decipher now if the `generateVariations` action is being triggered. Previously we were not able to differentiate if it was `loading` the variations or `generating` them.